### ISSUE TITLE
[rewrite] (recycled?) href links send address bar to the wrong URL

### ISSUE THREAD

**Issue by spacejack** [lhorie/mithril.js#1342]
### Description

Under certain conditions, when assigning `a href` links using `oncreate: m.route.link` the `href` value will render correctly, however when clicking the link, the browser is sent to an incorrect URL.

### Steps to Reproduce:

Fiddle here:

https://jsfiddle.net/yL3ogtLs/

Easier to observe when not in an iframe:

http://www.spacejack.ca/mithril-routebug/

Must follow these steps to reproduce the bug:

1. Click the "Cats" link
2. Click the "Dogs" link
3. Click one of the items in the list. If you click `/dogs/1` the browser will be sent to `/cats/1`

Inspect the anchor element in the dogs list before clicking to verify that the href value was rendered correctly.

### Expected:

Browser address bar should be set to the href url value.

### Actual:

Browser address bar set to a different url.

### Workaround:

Can set the href like:

	m('a', {href: '#!/dogs/1'})

but you lose the ability to change routing methods.

**Comment by pygy** [comment]
IIRC when this was discussed in the chat, it was mentioned that removing [this line](https://github.com/lhorie/mithril.js/blob/5b8821fb2e598cf303c3f880357e4f551e2f1820/render/tests/test-component.js#L652) in the tests also triggered related errors.

**Comment by sdougbrown** [comment]
fwiw, setting a unique key on the `a` element is another workaround, but it's a pretty big gotcha to require that...
```
m('a', {
  href: '/'+entityName+'/'+row.id,
  key: entityName+'_'+row.id,
  oncreate: m.route.link
},  row.text)
```

**Comment by lhorie** [comment]
Fixed in 40251c6c16f29473d3ab0ecbad4293978d9b073c

**Comment by spacejack** [comment]
Awesome!! Thank you.
